src/xrc/xmlres.cpp

```diff
@@ -1892,36 +1938,110 @@ wxBitmap wxXmlResourceHandlerImpl::GetBitmap(const wxXmlNode* node,
         return wxNullBitmap;
     }
-    wxImage img(*(fsfile->GetStream()));
-    delete fsfile;
-#else
-    wxImage img(name);
-#endif
 
-    if (!img.IsOk())
+    /* If the bitmap is specified as stock item, query wxArtProvider for it: */
+    wxString art_id, art_client;
+    if ( GetStockArtAttrs(node, defaultArtClient,
+                          art_id, art_client) )
+    {
+        wxBitmapBundle stockArt(wxArtProvider::GetBitmapBundle(art_id, art_client, size));
+        if ( stockArt.IsOk() )
+            return stockArt;
+    }
+
+    wxBitmapBundle bitmapBundle;
+    wxString paramValue = GetParamValue(node);
+    bool isBitmaps = paramValue.Contains(";");
+    bool isSVG = paramValue.EndsWith(".svg");
+    if ( isBitmaps && isSVG )
     {
         ReportParamError
         (
-            node->GetName(),
-            wxString::Format("cannot create bitmap from \"%s\"", name)
+            param,
+            "may contain either one svg file or a list of files separated by ';'"
         );
-        return wxNullBitmap;
     }
-    if (!(size == wxDefaultSize)) img.Rescale(size.x, size.y);
-    return wxBitmap(img);
+    else if (isSVG)
+    {
+        // it is a bundle from svg file
+        wxString svgDefaultSizeAttr = node->GetAttribute("default_size", "");
+        if ( svgDefaultSizeAttr.empty() )
+        {
+            ReportParamError
+            (
+                param,
+                "'default_size' attribute required with svg file"
+            );
+        }
+        else
+        {
+#ifdef wxHAS_SVG
+            wxSize svgDefaultSize = ParseStringInPixels(this, param, svgDefaultSizeAttr, wxDefaultSize);
+#if wxUSE_FILESYSTEM
+            wxFSFile* fsfile = GetCurFileSystem().OpenFile(paramValue, wxFS_READ | wxFS_SEEKABLE);
+            if (fsfile == NULL)
+            {
+                ReportParamError
+                (
+                    param,
+                    wxString::Format("cannot open SVG resource \"%s\"", paramValue)
+                );
+            }
+            else
+            {
+                wxInputStream* s = fsfile->GetStream();
+                const size_t len = static_cast<size_t>(s->GetLength());
+                wxCharBuffer buf(len);
+                char* const ptr = buf.data();
+
+                if (s->ReadAll(ptr, len))
+                {
+                    bitmapBundle = wxBitmapBundle::FromSVG(ptr, svgDefaultSize);
+                }
+                delete fsfile;
+            }
+#else
+            bitmapBundle = wxBitmapBundle::FromSVGFile(paramValue, svgDefaultSize);
+#endif
+#else // !wxHAS_SVG
+            ReportParamError
+            (
+                param,
+                "SVG bitmaps are not supported in this build of the library"
+            );
+#endif // wxHAS_SVG/!wxHAS_SVG
+        }
+    }
+    else
+    {
+        // it is a bundle from bitmaps
+        wxVector<wxBitmap> bitmaps;
+        wxArrayString pathes = wxSplit(paramValue, ';');
```

Comment:
Two things again: first, there is a typo in the variable name and second, and more importantly, I don't think we want to allow escaping semicolons with backslashes here, which can be done by default, as backslashes can occur naturally in paths, at least under MSW, so:
```suggestion
        wxArrayString paths = wxSplit(paramValue, ';', '\0');
```
Alternatively, perhaps we could `';'` itself as the escape to allow escaping it inside the file names by just doubling them? If this works -- could you please test it? -- we probably should do it, and then document it.